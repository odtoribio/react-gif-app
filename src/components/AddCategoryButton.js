import React, { useState } from 'react'
import PropTypes from 'prop-types'  

export const AddCategoryButton = ({ addCategory }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputValue = (e) => {
    setInputValue(e.target.value)
  }

  const handleOnSubmit = (e) => {
    e.preventDefault()

    if (inputValue){
      addCategory(inputValue)
      setInputValue('')
    }
  }
  
  return (
    <form onSubmit={ handleOnSubmit }>
      <input 
        type="text"
        value={ inputValue }
        onChange= { handleInputValue }
      />
    </form>
  )
}

AddCategoryButton.propTypes = {
  addCategory: PropTypes.func.isRequired,
}
