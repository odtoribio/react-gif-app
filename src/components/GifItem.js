import React from 'react'
import PropTypes from 'prop-types';

export const GifItem = (props) => {
  const { title, url } = props;
  return (
    <div className="card animate__animated animate__zoomIn">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}


GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
