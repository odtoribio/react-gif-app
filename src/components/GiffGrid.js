import React from 'react';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GiffGrid = ( { category } ) => {

  const { data:gifs, loading } = useFetchGifs(category)

  return (
    <>
      <h3>{ category }</h3>
      { loading && 'Loading...' }
      <div className="card-grid">
          {
            gifs.map( gif => (
              <GifItem 
                key={gif.id} 
                gif={ gif }
              />
            ))
          }
      </div>
    </>
  )
}
