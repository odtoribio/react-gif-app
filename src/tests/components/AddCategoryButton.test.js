import React from 'react';
import {shallow} from 'enzyme';
import { AddCategoryButton } from '../../components/AddCategoryButton';

describe('Test AddCategoryButton component', () => {

  const addCategory = jest.fn()
  let  AddCategoryButtonComponent = shallow(<AddCategoryButton addCategory={addCategory} />)

  beforeEach( () => {
    jest.clearAllMocks()
    AddCategoryButtonComponent = shallow(<AddCategoryButton addCategory={addCategory} />)
  })

  test('should show component correctly', () => {
    expect(AddCategoryButtonComponent).toMatchSnapshot()
  })

  test('should change input text', () => {

    const textInput = AddCategoryButtonComponent.find('input');
    textInput.simulate('change', { target: { value: "Hola mundo"}})
    
  })

  test('should not post the information on submit', () => {

    AddCategoryButtonComponent.find('form').simulate('submit', { preventDefault: () => {}})
    expect(addCategory).not.toHaveBeenCalled()

  })

  test('should call addCategory and clean the text input', () => {
    
    const input = AddCategoryButtonComponent.find('input').simulate('change', { target: { value: "Test input" } })
    AddCategoryButtonComponent.find('form').simulate('submit', { preventDefault: () => {}})
    const inputValue = input.prop('value');

    expect(addCategory).toHaveBeenCalled();
    expect(inputValue).toBe('');

  })
  
})
