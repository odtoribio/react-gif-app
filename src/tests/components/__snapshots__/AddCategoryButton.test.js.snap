// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddCategoryButton component should show component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
